src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -2367,6 +2390,38 @@ private void HandleResult(int node, BuildResult result)
                 configuration.ProjectTargets ??= result.ProjectTargets;
             }
 
+            // Only report results to the project cache services if it's the result for a build submission.
+            // Note that graph builds create a submission for each node in the graph, so each node in the graph will be
+            // handled here. This intentionally mirrors the behavior for cache requests, as it doesn't make sense to
+            // report for projects which aren't going to be requested. Ideally, *any* request could be handled, but that
+            // would require moving the cache service interactions to the Scheduler.
+            if (_buildSubmissions.TryGetValue(result.SubmissionId, out BuildSubmission buildSubmission))
+            {
+                // The result may be associated with the build submission due to it being the submission which
+                // caused the build, but not the actual request which was used with the build submission. Ensure
+                // only the actual submission's request is considered.
```

Comment:
I don't get it, can you elaborate?

Reply:
Sure, I'll clarify the comment but basically if a build submission causes other build requests to fire off (ie dependencies), then we'll get results for those and it will have the submission if that kicked the whole thing off, but what we're looking for here is just the result for the "top level" request.

In a well-described graph, all results would be for top-level requests :)